unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -43,12 +43,13 @@
    * Finally is the unit name, ends with an optional "s".
    */
   private static String unitRegex(String unit) {
-    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?)?";
+    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?\\s*)?";
```

Comment:
One possible thing that could happen with unsafe for interval is to support reading directly from a memory address rather than materializing it as a long + an int.